datastore/kubernetes.go

```diff
@@ -430,9 +430,35 @@ func (s *DataStore) IsKubeNodeUnschedulable(nodeName string) (bool, error) {
 	if err != nil {
 		return false, err
 	}
+
+	clusterAutoscalerEnabled, err := s.GetSettingAsBool(types.SettingNameKubernetesClusterAutoscalerEnabled)
```

Comment:
I think we should just allow the drain w/ cordon case triggered by cluster autoscaler to avoid unexpected following replica scheduling, so that said we just keep the current function implementation as is instead of respecting `ToBeDeletedByClusterAutoscaler` which also include the drain w/o cordon case. WDYT?

Reply:
After discussing with @c3y1huang , I think there are no obvious ways to let users configure `cordon before drain` in cloud provider cluster autoscaler. 

Also, the specific taint added by CA actually means the node is marked to scale down, so it indirectly means it's unscheduled for Longhorn. Let's keep the change as is first. 

ref: https://github.com/kubernetes/ingress-gce/issues/595

Reply:
After talking to @innobead ,

There is an option flag [cordon-node-before-terminating](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/main.go#L185). If this option is enabled, CA will [mark the node spec unschedulable](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/utils/deletetaint/delete.go#L120-L122).
However, regardless of the option flag, CA will taint the node with `ToBeDeletedByClusterAutoscaler`.

There could be a possible corner case when the user manually tolerates all taints and the replica could still rebuild onto the to be deleted node.

We think this can be tackled later if users have further concerns.